### PR TITLE
feat: separate the service from the bridge controller

### LINKED ISSUE #282: Split db in read and write

### PR DESCRIPTION
**Pull request by cool-develope**
Closes #282 
### Reviewers

Main reviewers:

- @ARR552 
- @arnaubennassar 

Codeowner reviewers:

- @tclemos

### CODE REVIEW

File: test/client/client.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (c RestClient) GetBridges(destAddr string, offset, limit uint) ([]*pb.Depos
 	if err != nil {
 		return nil, 0, err
 	}
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	bodyBytes, err := io.ReadAll(resp.Body)
```
Comment (ARR552) [inline test/client/client.go:123]: What is the difference?

File: config/default.go [hunk 2]
```diff
@@ -6,13 +6,13 @@ const DefaultValues = `
...
 
-[Database]
+[SyncDB]
 Database = "postgres"
 User = "test_user"
 Password = "test_password"
 Name = "test_db"
 Host = "localhost"
-Port = "5433"
+Port = "5435"
```
Comment (ARR552) [inline config/default.go:15]:
```suggestion
Port = "5432"
```
I think it is better to use the default config for db as well

File: config/default.go [hunk 3]
```diff
@@ -33,4 +33,25 @@ Height = 32
...
+DefaultPageLimit = 25
+MaxPageLimit = 100
+BridgeVersion = "v1"
+    [BridgeServer.DB]
+	Database = "postgres"
+    User = "test_user"
+    Password = "test_password"
+    Name = "test_db"
+    Host = "localhost"
+    Port = "5435"
```
Comment (ARR552) [inline config/default.go:45]:
```suggestion
    Port = "5432"
```

File: config/config.debug.toml [hunk 4]
```diff
@@ -27,6 +27,17 @@ Height = 32
...
+DefaultPageLimit = 25
+MaxPageLimit = 100
+BridgeVersion = "v1"
+    [BridgeServer.DB]
+    Database = "postgres"
+    User = "test_user"
+    Password = "test_password"
+    Name = "test_db"
+    Host = "localhost"
+    Port = "5435"
```
Comment (ARR552) [inline config/config.debug.toml:39]: Why 5435?? Using this value, I think we should modify the docker compose file

**Review by ARR552 — CHANGES_REQUESTED** [review]
I see that we have a duplication in getProof from API and getSiblings in mt. Is there any way to avoid it?
You could take the opportunity to fix the migration files to make them compatible with public

File: config/config.debug.toml (hunk 4 above)
Comment (cool-develope) [reply config/config.debug.toml:39]: for the `localhost`, we should use the binding port, right?

File: test/client/client.go (hunk 1 above)
Comment (cool-develope) [reply test/client/client.go:123]: `golint` suggests it, looks like `ioutil` is deprecated

**Comment by cool-develope** [comment]
> I see that we have a duplication in getProof from API and getSiblings in mt. Is there any way to avoid it? You could take the opportunity to fix the migration files to make them compatible with public

`getSiblings` will look different, since we will keep siblings in memory, right?

File: config/default.go [hunk 5]
```diff
@@ -37,15 +37,14 @@ DefaultPageLimit = 25
 MaxPageLimit = 100
 BridgeVersion = "v1"
     [BridgeServer.DB]
-	Host = "zkevm-bridge-db"
+    Database = "postgres"
     User = "test_user"
     Password = "test_password"
     Name = "test_db"
-    Host = "localhost"
+    Host = "zkevm-bridge-db"
```
Comment (ARR552) [inline config/default.go:44]: Why don't you modify this on the config file?

**Review by ARR552 — APPROVED** [review]

**Pull request merged**